pkg/controller/istio/istio_controller.go

```diff
@@ -0,0 +1,465 @@
+	r := newReconciler(mgr, opts)
+
+	c, err := ctrlruntime.NewController("istio-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return fmt.Errorf("failed to create istio-controller: %w", err)
+	}
+
+	// Watch for changes to primary resource Istio
+	err = c.WatchObject(&operatorv1.Istio{}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		log.V(5).Info("Failed to create Istio watch", "err", err)
+		return fmt.Errorf("istio-controller failed to watch primary resource: %v", err)
+	}
+
+	if err = utils.AddInstallationWatch(c); err != nil {
+		log.V(5).Info("Failed to create network watch", "err", err)
+		return fmt.Errorf("istio-controller failed to watch Tigera network resource: %v", err)
+	}
+
+	err = c.WatchObject(&crdv1.FelixConfiguration{}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("istio-controller failed to watch FelixConfiguration resource: %w", err)
+	}
+
+	if err = utils.AddPeriodicReconcile(c, utils.PeriodicReconcileTime, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("istio-controller failed to create periodic reconcile watch: %w", err)
+	}
+
+	return nil
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, opts options.AddOptions) *ReconcileIstio {
+	r := &ReconcileIstio{
+		Client: mgr.GetClient(),
+		scheme: mgr.GetScheme(),
+		status: status.New(mgr.GetClient(), "istio", opts.KubernetesVersion),
+	}
+
+	r.status.Run(opts.ShutdownContext)
+	return r
+}
+
+// ReconcileIstio reconciles a Istio object
+type ReconcileIstio struct {
+	client.Client
+	scheme *runtime.Scheme
+	status status.StatusManager
+}
+
+func (r *ReconcileIstio) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.V(1).Info("Reconciling Istio")
+
+	if request.NamespacedName != utils.DefaultInstanceKey {
+		reqLogger.V(1).Info("Istio resource named %q is not recognised, name it %q",
+			request.Name, utils.DefaultInstanceKey.Name)
+		return reconcile.Result{}, nil
+	}
+
+	// Get the Istio CR.
+	instance := &operatorv1.Istio{}
+	err := r.Get(ctx, utils.DefaultInstanceKey, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			reqLogger.Info("Istio object not found")
+			r.status.OnCRNotFound()
+			return reconcile.Result{}, nil
+		}
+		r.status.SetDegraded(operatorv1.ResourceReadError,
+			fmt.Sprintf("Error querying for Istio CR: failed to get Istio %q", utils.DefaultInstanceKey),
+			err, reqLogger)
+		return reconcile.Result{}, err
+	}
+	updateDefaults(instance)
+	r.status.OnCRFound()
+
+	// SetMetaData in the TigeraStatus such as observedGenerations.
+	defer r.status.SetMetaData(&instance.ObjectMeta)
+
+	defer r.setCondition(ctx, instance, reqLogger)
+
+	setCurrentCondition(instance, v1.ComponentProgressing, reasonInfo, "Deploying Tigera Istio resources")
```

Comment:
I don't think you should need any of these `setCurrentCondition` calls - we can handle this entirely by inheriting conditions from the TigeraStatus, which is a pattern we do throughout the other controllers. For example: 

https://github.com/tigera/operator/blob/cdb970b7f2a9d05f388d33a4bb3241d6f76f0c0f/pkg/controller/installation/core_controller.go#L785-L797


This reduces the amount of special handling code we need, and also avoids us e.g., toggling conditions every time we reconcile (we don't need to set Progressing every time Reconcile is called - sometimes it is called and nothing changes!)